syntax = "proto3";

package statistics;

message Empty {}

message TaskStatRequest {
    int32 id = 1;
}

message TaskStatResponse {
    int32 likes = 1;
    int32 views = 2; 
}

message PopularTaskRequest {
    string sort_type = 1;
}

message TaskRsp {
    int32 id = 1;
    string author_id = 2;
    int32 count = 3;
}

message PopularTaskResponse {
    repeated TaskRsp popular_tasks = 1;
}

message UserRsp {
    int32 author_id = 1;
    int32 likes = 2;
}

message PopularUserResponse {
    repeated UserRsp popular_users = 1;
}

service StatisticService {
    rpc GetTaskStatistic(TaskStatRequest) returns (TaskStatResponse);
    rpc GetPopularTask(PopularTaskRequest) returns (PopularTaskResponse);
    rpc GetPopularUser(Empty) returns (PopularUserResponse);
}
